src/model/network-legacy.cpp

```diff
@@ -149,8 +149,12 @@ bool LegacyNetwork::DistributedMulticastSupported() const
 
 EvalStatus LegacyNetwork::Evaluate(const tiling::CompoundTile& tile,
                                  const double inner_tile_area,
-                                 const bool break_on_failure)
+                                 const bool break_on_failure,
+                                 const bool reduction)
```

Comment:
So the "reduction" flag will be invoked on a per-mapping basis. The availability of this flag implies that the underlying network has a MUX that allows using or bypassing the spatial adder. But what are the semantics if there's an address collision (indicated by a "multicast factor" in the tileinfo), and the reduction flag is not set? Couple of other thoughts: (a) at the moment we don't have a distinction between design-time network connections and config-time LI channel setup - we really do everything at design (i.e. topology construction) time. This flag is more forward-looking, because it allows mapping-specific selection of which way to swing the MUX. This is fine, but the rest of the infrastructure has to catch up to this level of flexibility as well. (b) Is there a generalization of this "reduction" operator? Are there different kinds of potential reductions? How would we expose them? Are there other transformations that the network could support other than classes of reduction?

Reply:
In fact, the reduction flag is no longer there, and I switch to use the ConnectionType for the original need (separating Read/Update traffic). I think that the reduction flag makes more sense when we want to use it for the mapper to decide "where" and "when" to perform the reduction. But that is a different feature from the original goal of these commits that separate the ReadFill/UpdateDrain traffic.